src/Sarif.Viewer.VisualStudio.Core/Services/DataService.cs

```diff
@@ -42,28 +50,43 @@ public void SendEnhancedResultData(Stream stream)
 
             if (sarifLog != null)
             {
-                this.SendEnhancedResultData(sarifLog);
+                return this.SendEnhancedResultData(sarifLog);
             }
+
+            return -1;
         }
 
         /// <inheritdoc/>
-        public void SendEnhancedResultData(SarifLog sarifLog)
+        public int SendEnhancedResultData(SarifLog sarifLog)
```

Comment:
>int [](http://example.com/codeflow?start=15&length=3)

Doesn't look like we can return this to the interop caller with the current interop code. It's not critical now, since Kyle is calling this directly, but we should keep it in mind for future investigation.

Reply:
is it because the API has to return Task<bool>?
can we have an out parameter to return it? E.g.
```
        public Task<bool> SendEnhancedResultDataAsync(SarifLog sarifLog, out int cookie)
        {
            return this.CallServiceApiAsync(ViewerDataServiceInterfaceName, (service) =>
            {
                cookie = service.SendEnhancedResultData(sarifLog);
                return true;
            });
        }
```

Reply:
I'm not sure. Maybe it can return Task<int>?

Reply:
updated return Task<int> pls take a look.